docs: switch util.inherits to class extends (#5677)
mochajs/mocha: docs/src/content/docs/explainers/third-party-reporters.mdx

---
description: Create custom reporters.

> title: Third-party reporters

Reviewer: Going to put this into its own PR

```suggestion
title: Third party reporters
```

> title: Third party reporters

Reviewer: While we're here, if it's OK :)

```suggestion
title: Third-party reporters
```

---

Mocha allows you to define custom third-party reporters within your own test suite, or by using npm modules. For example if "lcov-reporter" was published to npm, you would simply add it to your package.json in `devDependencies` and use `--reporter lcov-reporter`.

:::note[New in v12.0.0]

> In Mocha 12, the extension logic is switched to modern `class`es instead of `util.inherits`.

Reviewer: ```suggestion
In Mocha 12, the extension logic is switched to modern `class ... extends` syntax instead of `util.inherits`.
```

Reviewer: Closing this as a nit

> If you're on Mocha 11 or older, use `mocha.util.inherits`.

Reviewer: Can/should we retain the full example for Mocha 11? I guess we can ignore it and Mocha 11 users wanting to create their first custom reporter will be able to reference the old docs, but maybe we could at least link to the old version of this page in source, or even link to this PR itself?

Author: We're ... what, a week away from launching v12? I think it's fine to switch to the new one. There's always the legacy site.

Author: I do like the link idea. working...

Reviewer: @JoshuaKGoldberg any update here?

Author: I've been procrastinating this PR 🙃. doing now...

Author: #6082 - prerequisite

:::

> Here is a minimalistic sample reporter for Mocha 12, which you can use by executing: `mocha --reporter my-reporter.js`

Reviewer: I'm still getting issues, can you share a full repro?

[mark-wiemer/hello-hello repo, /packages/mocha/packages/repro at 48abaed](https://github.com/mark-wiemer/hello-hello/tree/48abaed505965bcb2670393c2e77290d4f516a1b/packages/mocha/packages/repro)

Reviewer: `export default` (both keywords necessary) fixes this :)
